### PR TITLE
Allow tagging customers

### LINKED ISSUE #3684: In order to identify special clients during service, I want to be able to associate tags or flags with clients
**In order to**  `achieve something`
To improve customer service I want to be able to identify users as 
1. VIP
2. ProblemClient
3. WeSkrewedUp

In such a way that later, when they create a new order, this special status can be communicated to the relevant users. 

**As a** `role`
Everyone

**I want** `something to be added or changed`
In a user´s profile page I want a clickable button that can identify them as a VIP, PC, or WSU
I DO NOT WANT THEM TO SEE THIS INFORMATION IN THEIR PROFILE

Also, should these categories be modifiable by the cooperative? If so, if would be easy to use the existing tags feature and the cooperative makes a tag associated with them.

HOWEVER then the visualization of those types of "Flags" (flags alert you to important information in real time that you might miss otherwise, tags are an organizational tool) must be different. maybe a bar instead of a circle. 

**Whereas currently** `this happens`
Nothing, we cannot do this

**Additional context**
image

### PR DESCRIPTION
**Pull request by AwesomeKuro**
Fixes #3684

### CODE REVIEW

File: src/Entity/User.php [hunk 1]
```diff
@@ -62,8 +64,11 @@
  * @UniqueEntity(fields={"usernameCanonical"}, errorPath="username")
  * @UniqueEntity("facebookId")
  */
-class User extends BaseUser implements JWTUserInterface, ChannelAwareInterface
+class User extends BaseUser implements TaggableInterface
```
Comment (alexsegura) [inline src/Entity/User.php:67]: Actually, we want to tag customers, not users. A person ordering always maps to a customer, but not necessarily to a user.

File: docker/php/php.ini [hunk 2]
```diff
@@ -1,5 +1,5 @@
 apc.enable_cli = 1
-date.timezone = Europe/Paris
+date.timezone = America/Monterrey
```
Comment (alexsegura) [inline docker/php/php.ini:2]: This shouldn't be commited

File: templates/admin/users.html.twig [hunk 3]
```diff
@@ -101,6 +101,7 @@
 <thead>
   <th>{% trans %}profile.email{% endtrans  %}</th>
   <th>{% trans %}profile.fullName{% endtrans  %}</th>
+  <th>{% trans %}profile.tags{% endtrans  %}</th>
```
Comment (alexsegura) [inline templates/admin/users.html.twig:104]: You can use `adminDashboard.tags.title` instead, it's already translated

File: app/config/services/twig.yml [hunk 4]
```diff
@@ -16,6 +16,10 @@ services:
     arguments:
       $assetsFilesystem: '@assets_filesystem'
 
+  AppBundle\Twig\TagsRuntime:
+    tags:
+        - { name: twig.tagsruntime }
+
```
Comment (alexsegura) [inline app/config/services/twig.yml:22]: This is useless, the service is autowired by Symfony.

File: templates/profile/profile.html.twig [hunk 5]
```diff
@@ -3,7 +3,15 @@
...
 </div>
+      
 
+{% if (user.isSuperAdmin) %}
+  <p><strong>{% trans %}profile.tags{% endtrans %}</strong>:
+    {% for tag in user.customer.tags|expand_tags %}
+      <span class="label label-default" style="background-color: {{ tag.color }};">{{ tag.slug }}</span>
+    {% endfor %}
+  </p>
+{% endif %}
```
Comment (alexsegura) [inline templates/profile/profile.html.twig:14]: - It's useless for admins to know about their own tags
- We are not using the "super admin" role in any case

File: src/Entity/Model/TaggableTrait.php [hunk 6]
```diff
@@ -20,25 +23,35 @@ public function getTaggableResourceClass()
...
     public function getTags()
     {
         if (null === $this->tags) {
             $this->tags = [];
         }
-
+        $log = new Logger('getTags');
+        $log->pushHandler(new StreamHandler('php://stdout', Logger::WARNING)); // <<< uses a stream
+        $log->warning('getTags');
+        $log->warning(print_r($this->tags, true));
```
Comment (alexsegura) [inline src/Entity/Model/TaggableTrait.php:36]: Please remove

File: src/Entity/Model/TaggableTrait.php [hunk 7]
```diff
@@ -20,25 +23,35 @@ public function getTaggableResourceClass()
...
      * @SerializedName("tags")
-     * @Groups({"task_create", "task_edit"})
+     * @Groups({"task_create", "task_edit", "admin_user_edit", "update_profile"})
      */
     public function setTags($tags)
     {
+        $log = new Logger('setTags');
+        $log->pushHandler(new StreamHandler('php://stdout', Logger::WARNING)); // <<< uses a stream
+        $log->warning('setTags');
+        $log->warning(print_r($tags, true));
```
Comment (alexsegura) [inline src/Entity/Model/TaggableTrait.php:49]: Please remove logging code

File: src/Form/UpdateProfileType.php [hunk 8]
```diff
@@ -146,6 +155,44 @@ function (FormEvent $event) use ($options, $isAdmin) {
...
+            //$log->warning(var_dump($user));
+            //$log->warning(var_dump($event));
+            $tags = $event->getData();
+            //$tags = explode(' ', $tags);
+
+            $user->setTags($tags);
+            //$user->getUser()->setTags($tags);
+        });
+
+        */
```
Comment (alexsegura) [inline src/Form/UpdateProfileType.php:195]: Please remove dead code

File: templates/profile/profile.html.twig [hunk 9]
```diff
@@ -4,6 +4,7 @@
   </a>
 </div>
 
+<p><strong>{% trans %}profile.tags{% endtrans %}</strong>: {{ user.tags|join(', ', ' and ') }}</p>
```
[the remark this answers is not in the document]
Comment (AwesomeKuro) [reply templates/profile/profile.html.twig]: Can i use an if to check if the user is an administrator?

**Comment by AdrienSICKLO** [comment]
image

This client is a VIP customer. The tag appear on the foodtech dashboard, but does not appear on the dispatch screen.

image

**Pull request merged**